x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -67,6 +70,14 @@ protected void closeInternal() {
 
     private final Set<EvictionListener> listeners = new HashSet<>();
 
+    /**
+     * Indicates whether the cache file has been synchronized with the storage device that contains it, since the last time data
+     * were written in cache (or since the creation of the file if no cached data have been written yet). An empty cache file is
+     * considered as fsynced (the initialization value is {@code true}) when it is created; and writing new data to the cache file
+     * will toggle the flag to {@code false}.
+     **/
+    private final AtomicBoolean fsynced = new AtomicBoolean(true);
```

Comment:
I wonder if calling this `needsFsync` (and reverting the boolean) would be slightly more intuitive?

With the current name, a reader would be inclined to think the boolean is only true when the file is fsync'ed, but this is not true (it is also still true while writing and completing the gap).

Reply:
Thanks, it makes sense. I pushed `d8fdaaa2aab25e78033906f2fa8c956a6e340441`.